### ISSUE TITLE
How I'm supposed to replace console.logs, setTimeouts or process.stdout.write?

### ISSUE THREAD

**Issue by Mdsp9070** [jonaskello/eslint-plugin-functional#152]
Actually is only a doubt. These expressions maybe cause side-effects, but how I'm supposed to replace them? I really need to set the rule off? But that will not open to another side-effects on the file?

**Comment by RebeccaStevens** [comment]
Write functions (e.g. `console.log`) are pretty much by definition not functional; therefore this lib will flag them by default. To allow them, you need to set an "ignore pattern" on the `no-expression-statement` rule.

For example:

```jsonc
"rules": {
  "no-expression-statement": ["error", {
    "ignorePattern": [
      "^console\\."  // Allow console.* functions.
    ]
  }]
}
```

The same can be done with `setTimeout` or alternatively, the "timeout ID" return by `setTimeout` can be assigned to a variable.

**Comment by Mdsp9070** [comment]
Oh, okay, now I understand it! Thank you!

**Comment by atmosbear** [comment]
> Write functions (e.g. `console.log`) are pretty much by definition not functional; therefore this lib will flag them by default. To allow them, you need to set an "ignore pattern" on the `no-expression-statement` rule.

Just want to add to anyone who found this question like I did that the correct one is "no-expression-statements" with an s haha. Took me a while to figure out why it wasn't working.

also I had to put it like this with the 'functional/' prefix:

```js
"functional/no-expression-statements": [
   "error",
   {
    "ignorePattern": [
     "console\\." // Allow console.* functions.
    ]
   }
  ],
```
